velox/vector/ComplexVector.cpp

```diff
@@ -809,31 +809,58 @@ VectorPtr RowVector::pushDictionaryToRowVectorLeaves(const VectorPtr& input) {
       wrappers, input->size(), input, input->pool());
 }
 
-void ArrayVectorBase::checkRanges() const {
-  std::unordered_map<vector_size_t, vector_size_t> seenElements;
-  seenElements.reserve(size());
+template <bool kHasNulls>
+vector_size_t ArrayVectorBase::nextNonEmpty(vector_size_t i) const {
+  while (i < size() &&
+         ((kHasNulls && bits::isBitNull(rawNulls(), i)) || rawSizes_[i] <= 0)) {
+    ++i;
+  }
+  return i;
```

Comment:
Shall we return std::optional if not found?

Reply:
Probably no need, we will need to check `i < size()` once again anyway and this won't save us anything in term of efficiency (it's even probably one more registry used) or code readability.